modules/tensor_mechanics/src/utils/GuaranteeConsumer.C

```diff
@@ -45,7 +45,7 @@ GuaranteeConsumer::hasGuaranteedMaterialProperty(const MaterialPropertyName & pr
     // If block materials exist, look if any issue the required guarantee
     if (warehouse.hasActiveBlockObjects(id))
     {
-      const std::vector<std::shared_ptr<Material>> & mats = warehouse.getActiveBlockObjects(id);
+      const std::vector<std::shared_ptr<MaterialBase>> & mats = warehouse.getActiveBlockObjects(id);
```

Comment:
This is an example why `InterfaceMaterial` have to be handled separately. They should not be included the active block objects. I believe it is the case with your code changes...